pages/admin/AdminHandler.php

```diff
@@ -119,11 +119,15 @@ public function initialize($request)
             ]);
         }
 
+        if (!Config::getVar('general', 'show_upgrade_warning')) {
+            return parent::initialize($request);
+        }
```

Comment:
@henriquejsfj the `VersionCheck::checkIfNewVersionExists()` must be always executed. So, the proper fix is to catch the `GuzzleHttp\Exception\ConnectException` on the call site.

Reply:
Obrigado @jonasraoni 
I was focused to make it work with setting the options, as the previous Issue solution, that I didn't realize that the proper fix is to really avoid throwing the fatal error hahah.